azext_iot/common/base_discovery.py

```diff
@@ -292,7 +292,7 @@ def get_target(
             )
 
         policy_name = kwargs.get("policy_name", "auto")
-        rg = resource.additional_properties.get("resourcegroup")
```

Comment:
Another way to fix this is to have the if statement depend on the version of the dps mgmt sdk installed... But this code change does not rely on that so I do like this better